tests/integration/reportinglib/1.check.in

```diff
@@ -4,18 +4,45 @@
 
 OK=0
 FAILED=1
+sonata_reports=@ENABLE_SONATA_REPORTS@
+reportinglib=@ENABLE_REPORTINGLIB@
 
-if [ -f test_1.bbp ] 
+if [ "$reportinglib" = "ON" ]
 then
-   #checking something with @reportinglib_BIN_DIR@/dump test_1.bbp
-   # for ascii files that will be using grep for instance.
-   nb_recorded=$(@reportinglib_somaDump@ test_1.bbp 1| wc -l)
-   if [ $nb_recorded -eq 100 ]; then
-     exit $OK
-   else
+  if [ -f test_1.bbp ]
+  then
+     #checking something with @reportinglib_BIN_DIR@/dump test_1.bbp
+     # for ascii files that will be using grep for instance.
+     nb_recorded=$(@reportinglib_LIB_DIR@/../usr/local/bin/somaDump test_1.bbp 1| wc -l)
+     # 101 because a white space
+     if [ $nb_recorded -eq 100 ]; then
+       exit $OK
+     else
+       echo "[ERROR] The number of steps recorded doesn't match. Test failed!" >&2
+       exit $FAILED
+     fi
+  else
+     echo "[ERROR] reportinglib soma file doesn't exist. Test failed!" >&2
      exit $FAILED
-   fi
-else
-   exit $FAILED
+  fi
 fi
 
+if [ "$sonata_reports" = "ON" ]
+then
+  if [ -f test_2.h5 ]
+  then
+     #checking something with @reportinglib_BIN_DIR@/dump test_1.bbp
+     # for ascii files that will be using grep for instance.
+     nb_recorded_sonata=$(h5dump -d /report/data -y -O test_2.h5 | wc -l)
```

Comment:
Are you sure `h5dump` is available in PATH? It should be found by CMake and use in this script with a variable like `@H5DUMP_EXECUTABLE@`